gtfs_realtime_translators/translators/njt_rail.py

```diff
@@ -0,0 +1,139 @@
+
+                    route_id = cls.__get_route_id(line=item_entry['LINE'],
+                                                  line_abbreviation=item_entry['LINEABBREVIATION'],
+                                                  origin=origin_and_destination[0],
+                                                  destination=origin_and_destination[1])
+
+                trip_update = TripUpdate.create(entity_id=str(idx + 1),
+                                                departure_time=departure_time,
+                                                scheduled_departure_time=scheduled_departure_time,
+                                                arrival_time=departure_time,
+                                                scheduled_arrival_time=scheduled_departure_time,
+                                                route_id=route_id,
+                                                route_short_name=route_short_name,
+                                                route_long_name=route_long_name,
+                                                route_color=route_color,
+                                                route_text_color=route_text_color,
+                                                stop_id=station_id,
+                                                stop_name=stop_name,
+                                                headsign=headsign,
+                                                track=track,
+                                                block_id=block_id,
+                                                agency_timezone='America/New_York')
+                trip_updates.append(trip_update)
+
+        return trip_updates
+
+    @classmethod
+    def __get_route_id(cls, **data):
+        """
+        This function resolves route_ids for NJT. The logic of determining a route_id based
+        on origin or destination is necessary to discern multiple routes that are mapped to the same line.
+
+        For instance, the North Jersey Coastline line operates two different routes. All trains with an
+        origin or destination of New York Penn Station should resolve to route_id 10 and the others route_id 11
+
+        :param data: keyword args containing data needed to perform the route logic
+        :return: route_id
+        """
+        route_id_lookup = {
+            'atlantic_city_ine': '1',
+            'montclair-boonton_line': None,
+            'main_line': '5',
+            'bergen_county_line': '6',
+            'morristown_line': '7',
+            'gladstone_branch': '8',
+            'northeast_corridor_line': '9',
+            'north_jersey_coast_line': None,
+        }
+
+        key = 'amtrak' if data['line_abbreviation'] == 'AMTK' else data['line'].replace(' ', '_').lower()
+        route_id = route_id_lookup.get(key, None)
+        if route_id is not None:
+            return route_id
+
+        def get_route_id_by_origin_or_destination(line_key, line_name, origin, destination):
```

Comment:
This looks fine. We should note these in our internal documentation for product/design to be aware.